src/app/navigation/types.ts

```diff
@@ -2,9 +2,9 @@ import { Location } from 'history';
 
 export type State = Location;
 
-export interface Match {
-  route: Route;
-  params: any;
+export interface Match<Params> {
+  route: Route<Params>;
+  params: Params;
```

Comment:
Oh, I didn't realize `Params` was a generic type. I was looking to see where `Params` was imported from. I realize we said no `I` in front of interfaces so maybe there's no need for generic types to begin with `T`? But I did notice `P` was used elsewhere in this diff... 


Any reason for choosing one (`Params`) over the other (`P`)?

(I think `P` helps clarify that it's a generic

Reply:
Also, what if the Match was enumerated with the different possible params used in the system? That would help prevent the `Match<any>` in `src/app/navigation/actions.ts`

Something like....

```ts
interface RouteAndParams<P> {
  route: Route<P>;
  params: P;
}

export type Match = RouteAndParams<undefined> | RouteAndParams<Params>
```

Reply:
you mean create a union type of the possible params? i can work on that, it seems possible.

Reply:
@philschatz i updated so the match typing is a bit more strict. it think its good now.

re the generic template names, i think convention is to use single letter names, but i had started using more verbose ones to make the hinting more useful when querying types from other files. the example you called out `Match<Params>` is kind of moot now because it became `Match<R extends AnyRoute>` which is pretty self explanatory. in general i'm going to try hard to make my generic parameters extend something so that you're limited in what you can pass in and you get a tip for what type of thing it should be. if the names are still too generic in some case i'll add a docblock, my editor at least will show me the docblock when i query the types.